Mage/src/main/java/mage/game/draft/DraftImpl.java

```diff
@@ -209,20 +219,53 @@ protected boolean pickCards() {
                 return false;
             }
             player.setPicking();
-            player.getPlayer().pickCard(player.getBooster(), player.getDeck(), this);
+            player.setBoosterNotLoaded();
         }
-        cardNum++;
+        setupBoosterLoadingHandle();
         synchronized (this) {
             while (!donePicking()) {
                 try {
-                    this.wait();
+                    this.wait(3000);
                 } catch (InterruptedException ex) {
                 }
             }
         }
+        cardNum++;
         return true;
     }
-
+    
+    protected void setupBoosterLoadingHandle() {
+        cancelBoosterLoadingHandle();
+        boosterLoadingCounter = 0;
+        boosterLoadingHandle = boosterLoadingExecutor.scheduleAtFixedRate(() -> {
+            try {
+                if (loadBoosters() == true) {
+                    cancelBoosterLoadingHandle();
+                } else {
+                    boosterLoadingCounter++;
+                }
+            } catch (Exception ex) {
+            }
```

Comment:
Shouldn't this do something with the exception?

Reply:
`scheduleAtFixedRate` automatically stops running if an exception is thrown. A logger message could be added, but I don't think it's necessary to add the logger to DraftImpl for this purpose.